### ISSUE TITLE
Workflows not defaulting to other sites when a site is in drain

### ISSUE THREAD

**Issue by jenimal** [dmwm/WMCore#8805]
Right now we are having massive failures in Workflows due to having to put Rome into drain because of ongoing site issues at Rome.
The site list for these workflows is fairly long, but they are only attempting to run at Rome, and then failing out, why aren't they just running at the other sites on the list?  Please see this set of workflows for an example... (just starting on the list for today)
pdmvserv_task_EXO-RunIISummer15GS-11399__v1_T_180910_180229_3947		100.0%	
pdmvserv_task_EXO-RunIISummer15GS-11398__v1_T_180910_180229_4968		100.0%	
pdmvserv_task_EXO-RunIISummer15GS-11394__v1_T_180910_180113_5242		100.0%	
pdmvserv_task_EXO-RunIISummer15GS-11396__v1_T_180910_180113_5823		100.0%	
pdmvserv_task_EXO-RunIISummer15GS-11382__v1_T_180910_175922_6718		100.0%	
pdmvserv_task_EXO-RunIISummer15GS-11383__v1_T_180910_175922_9611		100.0%	
pdmvserv_task_EXO-RunIISummer15GS-11385__v1_T_180910_175916_1603		100.0%	
pdmvserv_task_EXO-RunIISummer15GS-11391__v1_T_180910_175915_3895		100.0%

**Comment by amaltaro** [comment]
Jen, first problem is related to an issue Seangchan is working on, see
https://github.com/dmwm/WMCore/pull/8806#issuecomment-421210417

which means, unfortunately vocms0250 has MaxRetries=0, instead the usual number of retries.
I just fixed it in vocms0250 and all the other agents up&running.

Second issue is an inconsistent site report in WMStats. The job detail table properly shows T2_IT_Rome (which is correct!), while the `State Transition` list wrongly reports "T3_US_Colorado".

Can you please clarify when Rome was put into drain?

**Comment by amaltaro** [comment]
Answering my own question, it was put into drain around this CERN timestamp:
2018-09-10 22:33:51

and checking the code again, the one-line change Seangchan has in his PR will fix this issue for good, even though it should no longer happen since I updated all agents configs.

**Comment by jenimal** [comment]
https://ggus.eu/?mode=ticket_info&ticket_id=136548  I updated the ggus ticket when I put it into drain so here is the comment
update#14 Jennifer Adelman-McCarthy | 2018-09-10 | 18:58 | Changed CC to cms-comp-ops-workflow-team@cern.ch;cms-comp-ops-site-support-team@cern.chPublic Diary:Putting the site into drain manually for production. Everytime the site goes back into production we high failure rates with 99303's and the attached error message. | Changed CC to cms-comp-ops-workflow-team@cern.ch;cms-comp-ops-site-support-team@cern.chPublic Diary:Putting the site into drain manually for production. Everytime the site goes back into production we high failure rates with 99303's and the attached error message.
Changed CC to cms-comp-ops-workflow-team@cern.ch;cms-comp-ops-site-support-team@cern.chPublic Diary:Putting the site into drain manually for production. Everytime the site goes back into production we high failure rates with 99303's and the attached error message.
